scripts/python-skywater-pdk/skywater_pdk/gds_to_svg/gds_to_svg.py

```diff
@@ -0,0 +1,241 @@
+    print('scalegrid 1 2',           file=ofile)
+    print('cif istyle vendorimport', file=ofile)
+    print('gds readonly true',       file=ofile)
+    print('gds rescale false',       file=ofile)
+    print('tech unlock *',           file=ofile)
+    print('cif warning default',     file=ofile)
+    print('set VDD VPWR',            file=ofile)
+    print('set GND VGND',            file=ofile)
+    print('set SUB SUBS',            file=ofile)
+    print('gds read ' + gdsfile,     file=ofile)
+
+
+def run_magic(destdir, tcl_path, input_techfile, d="null"):
+    """
+    Runs magic to generate layout files.
+
+    Parameters
+    ----------
+    destdir: destination directory
+    tcl_path: path to input TCL file
+    input_techfile: path to the technology file
+    d: Workstation type, can be NULL, X11, OGL or XWIND
+    """
+    cmd = [
+        'magic',
+        '-nowrapper',
+        '-d'+d,
+        '-noconsole',
+        '-T', input_techfile,
+        os.path.abspath(tcl_path)
+    ]
+    with open(tcl_path.replace(".tcl", ".sh"), "w") as f:
+        f.write("#!/bin/sh\n")
+        f.write(" ".join(cmd))
+    mproc = subprocess.run(
+        cmd,
+        stdin=subprocess.DEVNULL,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        cwd=destdir,
+        universal_newlines=True)
+    assert mproc.stdout
+    max_cellname_width = 0
+    output_by_cells = [('', [])]
+    fatal_errors = []
+    for line in mproc.stdout.splitlines():
+        cifwarn = ('CIF file read warning: Input off lambda grid by 1/2; ' +
+                   'snapped to grid')
+        if line.startswith(cifwarn):
+            continue
+        m = FATAL_ERROR.match(line)
+        if m:
+            fatal_errors.append(line)
+        m = READING_REGEX.match(line)
+        if m:
+            cell_name = m.group(1)
+            max_cellname_width = max(max_cellname_width, len(cell_name))
+            output_by_cells.append((cell_name, []))
+        output_by_cells[-1][-1].append(line)
+    for cell, lines in output_by_cells:
+        prefix = "magic " + cell.ljust(max_cellname_width) + ':'
+        for line in lines:
+            is_error = 'rror' in line
+            if superdebug or (debug and is_error):
+                print(prefix, line)
+    assert not mproc.stderr, mproc.stderr
+    if mproc.returncode != 0 or fatal_errors:
+        if fatal_errors:
+            msg = ['ERROR: Magic had fatal errors in output:'] + fatal_errors
+        else:
+            msg = ['ERROR: Magic exited with status ' + str(mproc.returncode)]
+        msg.append("")
+        msg.append(" ".join(cmd))
+        msg.append('='*75)
+        msg.append(mproc.stdout)
+        msg.append('='*75)
+        msg.append(destdir)
+        msg.append(tcl_path)
+        msg.append('-'*75)
+        msg.append(msg[0])
+        raise SystemError('\n'.join(msg))
+    return output_by_cells
+
+
+def convert_to_svg(input_gds, input_techfile, output=None):
```

Comment:
This should probably be split into two functions, one to generate the original SVG from magic. The second one to cleanup the SVG from magic and write the final output.

Reply:
I subdivided it into separate methods